### PR TITLE
⚡ Optimize evalPopulationStdDevDecimal using Welford's algorithm

### PR DESCRIPTION
**Pull request by suyashkumar**
💡 **What:** Replaced the multi-pass implementation of Population Standard Deviation for Decimals with a single-pass Welford's algorithm.

🎯 **Why:** The previous implementation iterated over the list multiple times (Count, Avg, then Variance calculation), leading to O(3N) complexity. The new implementation is O(N) and single-pass, which is more efficient for large lists.

📊 **Measured Improvement:**
*   **Baseline:** ~6.17 ms/op
*   **Optimized:** ~4.05 ms/op
*   **Improvement:** ~34% speedup on a list of 10,000 decimals.

Also added comprehensive tests for `PopulationStdDev` covering edge cases.

---
*PR created automatically by Jules for task [16765708447522369412](https://jules.google.com/task/16765708447522369412) started by @suyashkumar*

### CODE REVIEW

**Comment by suyashkumar** [comment]
Can't seem to change this to draft, but no need for review yet, will go over a few of these to verify things first.

File: tests/enginetests/benchmark_stddev_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package enginetests
```
Comment (suyashkumar) [inline tests/enginetests/benchmark_stddev_test.go:15]: Please put the Apache 2.0 copyright block at the top of any new files, you can find this in other files in this repo.

**Comment by evan-gordon** [comment]
Overall LGTM.

Just curious, but does Jules have an option for squashing commits we could use in the future?
